src/auto-complete.jsx

```diff
@@ -64,7 +64,13 @@ const AutoComplete = React.createClass({
       open: this.props.open,
     };
   },
-
+  componentWillReceiveProps: function(nextProps) {
+    if (this.state.searchText !== nextProps.searchText) {
```

Comment:
no.. imagine this:

Render component with props: ""
Type into the component: "test"

At this point, props is "" and state is "test". Since the props don't get updated, we need to check with the latest value available, which is state. Does that make sense?